consensus/src/liveness/round_proposer_election.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::liveness::proposer_election::ProposerElection;
+use consensus_types::common::{Author, Round};
+
+use std::collections::HashMap;
+
+/// The round proposer maps a round to author(s)
+pub struct RoundProposer {
+    // A pre-defined map specifying proposers per round
+    proposers: HashMap<Round, Author>,
+    // Default proposer to use if proposer for a round is unspecified.
+    // We hardcode this to the first proposer
+    default_proposer: Author,
+}
+
+impl RoundProposer {
+    pub fn new(proposers: &HashMap<Round, Author>, default_proposer: Author) -> Self {
```

Comment:
just pass the HashMap instead of reference?

Reply:
Sounds good, done.